improver_tests/regrid/test_RegridWithLandSeaMask.py

```diff
@@ -0,0 +1,359 @@
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Unit tests for the RegridWithLandSeaMask class"""
+# set up a special data set and corresponding land-sea mask info
+# set up target grid and its land-sea mask info
+# it is designed to cover different scenarios for regridding with land-sea
+# the regridding reference results are manually checked for different methods
+# not using "set_up_variable_cube" because of different spacing at lat/lon
+
+import iris
+import numpy as np
+from iris.coords import DimCoord
+from iris.cube import Cube
+
+from improver.regrid.bilinear import basic_indexes
+from improver.regrid.grid import calculate_input_grid_spacing, latlon_from_cube
+from improver.regrid.landsea import RegridLandSea
+
+
+# function for creating cube from data, lats, lons
+def create_cube(cube_array, lats, lons, name, unit):
+    """
+    create a simple cube from data and lat/lon coords
+    args:
+        cube_array (numpy.ndarray):
+            data value (two dimensional)
+        lats ( numpy.ndarray):
+            latitude array
+        lons ( numpy.ndarray):
+            longitude array
+        name (str):
+            variable name
+        unit (str):
+            variable unit
+    return:
+         cube_v(iris.cube.Cube):
+             new-created cube
+    """
+    cube_v = Cube(cube_array)
+    # assume name
+    cube_v.var_name = name
+    # cube_v.standard_name= name
+    cube_v.units = unit
+
+    coord_lat = DimCoord(
+        lats,
+        standard_name="latitude",
+        units="degrees",
+        coord_system=iris.coord_systems.GeogCS(6371229),
+    )
+    coord_lon = DimCoord(
+        lons,
+        standard_name="longitude",
+        units="degrees",
+        coord_system=iris.coord_systems.GeogCS(6371229),
+    )
+
+    cube_v.add_dim_coord(coord_lat, 0)
+    cube_v.add_dim_coord(coord_lon, 1)
+
+    return cube_v
+
+
+def define_source_target_grid_data():
+    """ define cube_in, cube_in_mask,cube_out_mask using assumed data """
+    # source (input) grid
+    in_lats = np.linspace(0, 15, 4)
+    in_lons = np.linspace(0, 40, 5)
+
+    # target (output) grid
+    out_lats = np.linspace(0, 14, 8)
+    out_lons = np.linspace(5, 35, 11)
+
+    # assume a set of nwp data
+    data = np.arange(20).reshape(4, 5).astype(np.float32)
+
+    # input grid mask info
+    in_mask = np.empty((4, 5), dtype=np.int)
+    in_mask[:, :] = 1
+    in_mask[0, 2] = 0
+    in_mask[2, 2:4] = 0
+    in_mask[3, 2:4] = 0
+
+    # output grid mask info
+    out_mask = np.empty((8, 11), dtype=np.int)
+    out_mask[:, :] = 1
+    out_mask[0, 4:7] = 0
+    out_mask[1, 5] = 0
+    out_mask[5:9, 4:10] = 0
+
+    out_mask[6, 6] = 1
+    out_mask[7, 6] = 1
+    out_mask[1, 0] = 0
+
+    cube_in = create_cube(data, in_lats, in_lons, "air_temperature", "Celsius")
```

Comment:
I would still prefer you to create cubes of the right shape using the standard set up cube functions and then just replaced the points in the x and y coordinates after the cubes have been created. cube.coord("latitude).points = new_points etc.  It really is important that the cubes are set up in a standard way. I don't think this is a big change for you to make.

Reply:
I did it in the way you suggested. It works after resetting bounds. Thanks.